src/status_im/contexts/wallet/account/view.cljs

```diff
@@ -28,6 +28,7 @@
       (let [{:keys [name color formatted-balance
                     watch-only?]} (rf/sub [:wallet/current-viewing-account])
             customization-color   (rf/sub [:profile/customization-color])]
+        (rn/use-unmount #(rf/dispatch [:wallet/close-account-page]))
```

Comment:
> Also fixes wallet/close-account-page not being called if back button was pressed (or gesture) instead of pressing on the X button

I think we need to check for X button on-press in the entire wallet/app